### PR TITLE
chore: rewrite non-web protocol for Electron tracing

### LINKED ISSUE #13289: [BUG] electron tracing with `snapshot: true` not very functional
**Context:**
- Playwright Version: `1.20.2`
- Operating System: macOS
- Node.js version: `16.x`
- Browser: `Chromium`

**Code Snippet**

This is from our use of playwright in VSCodes Electron smoke tests, so I cannot really share a snippet. We setup playwright for Electron here (not using `snapshot: true`, `sources: true` in that link, but had it enabled for the attached trace): 

https://github.com/microsoft/vscode/blob/2ada17080cd3e351e093e5fce40d36f16db3124a/test/automation/src/playwrightElectronDriver.ts#L30-L61

**Describe the bug**

See attached trace and open it on https://trace.playwright.dev/. Try to operate in the DOM and notice how that is not possible. Screenshots seem to work fine though.

image

[playwright-trace-1-searches_for_body___checks_for_correct_result_number.zip](https://github.com/microsoft/playwright/files/8408238/playwright-trace-1-searches_for_body___checks_for_correct_result_number.zip)

//cc @pavelfeldman

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/13289

### CODE REVIEW

File: packages/trace-viewer/src/snapshotRenderer.ts [hunk 1]
```diff
@@ -273,3 +274,33 @@ function snapshotScript() {
...
+    href = href.substring(kLegacyBlobPrefix.length);
+
+  try {
+    const url = new URL(href);
+    // Sanitize URL.
+    if (url.protocol === 'javascript:')
+      return 'javascript:void(0)';
+
+    // Pass through if possible.
+    const isBlob = url.protocol === 'blob:';
```
Comment (dgozman) [inline packages/trace-viewer/src/snapshotRenderer.ts:294]: Can just  remove "blob" from the list, and everything should work?

File: packages/trace-viewer/src/snapshotServer.ts [hunk 2]
```diff
@@ -103,13 +100,3 @@ declare global {
     showSnapshot: (url: string, point?: Point) => Promise<void>;
   }
 }
-
-function removeHash(url: string) {
```
Comment (dgozman) [inline packages/trace-viewer/src/snapshotServer.ts:107]: Why dropped this one? We intercept without a hash, but page requests with a hash, so we make sure url matches.

**Review by dgozman — APPROVED** [review]

File: packages/trace-viewer/src/snapshotRenderer.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/trace-viewer/src/snapshotRenderer.ts:294]: Done.

File: packages/trace-viewer/src/snapshotServer.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/trace-viewer/src/snapshotServer.ts:107]: Ouch

**Pull request merged**